src/realm-settings/RealmSettingsSection.tsx

```diff
@@ -149,6 +149,15 @@ const RealmSettingsHeader = ({
   );
 };
 
+const sortByPriority = (components: ComponentRepresentation[]) => {
+  const sortedComponents = [...components].sort(
+    // sort in descending order
+    (a, b) => Number(b.config?.priority) - Number(a?.config?.priority || 0)
+  );
```

Comment:
I think we can catch all the edge cases here by casting the values to a `number` first and then checking if those values are `NaN` (defaulting to 0 if they are). Since both `Number([])` and `Number(undefined)` will produce `NaN` only 'good' values will pass through here.

I think the type is actually wrong and this value specifically would always be of type `string`, but it's probably good to cover possible edge cases if we are not entirely sure.
```suggestion
  const sortedComponents = [...components].sort((a, b) => {
    const priorityA = Number(a.config?.priority);
    const priorityB = Number(b.config?.priority);

    return (
      (!isNaN(priorityB) ? priorityB : 0) - (!isNaN(priorityA) ? priorityA : 0)
    );
  });
```

Reply:
defaulting to 0 here is a good idea - thank you @jonkoops